geoh5py/objects/grid2d.py

```diff
@@ -141,6 +140,90 @@ def centroids(self) -> np.ndarray | None:
 
         return self._centroids
 
+    def copy_from_extent(
+        self,
+        extent: list[float] | np.ndarray,
+        parent=None,
+        copy_children: bool = True,
+        clear_cache: bool = False,
+        **kwargs,
+    ):
+        """
+        Function to copy an entity to a different parent entity.
+
+        :param extent: Extent of the copied entity.
+        :param parent: Target parent to copy the entity under. Copied to current
+            :obj:`~geoh5py.shared.entity.Entity.parent` if None.
+        :param copy_children: (Optional) Create copies of all children entities along with it.
+        :param clear_cache: Clear array attributes after copy.
+        :param kwargs: Additional keyword arguments to pass to the copy constructor.
+
+        :return entity: Registered Entity to the workspace.
+        """
+        if isinstance(extent, list):
+            extent = np.vstack(extent)
```

Comment:
**issue:** does not convert flat list `list[float]` to a 2 dimension array

maybe not need to have the list type for this param...

```
>>> import numpy
>>> numpy.vstack([0,1,2,3,4,5])
array([[0],
       [1],
       [2],
       [3],
       [4],
       [5]])
```

Reply:
Was mainly for convenience of supplying [pt_1, pt_2], instead of forcing to be a numpy array... I just think I need to change the type hint to list[list]

Reply:
possibly, as long as tests are covering both the numpy and the list input